DMCompiler/DM/DMProc.cs

```diff
@@ -245,16 +246,46 @@ private class DMProcScope {
             WriteLabel(jumpTo);
         }
 
-        public void Break() {
-            Jump(_loopStack.Peek() + "_end");
+        public void Break(DMASTIdentifier label = null) {
+            if (label is not null)
+            {
+                Jump(label.Identifier + "_end");
+            }
+            else
+            {
+                Jump(_loopStack.Peek() + "_end");
+            }
         }
 
         public void BreakIfFalse() {
             JumpIfFalse(_loopStack.Peek() + "_end");
         }
 
-        public void Continue() {
-            Jump(_loopStack.Peek() + "_continue");
+        public void Continue(DMASTIdentifier label = null) {
+            // TODO: Clean up this godawful label handling
+            if (label is not null)
+            {
+                var codeLabel = label.Identifier + "_codelabel";
+                if (!_labels.ContainsKey(codeLabel))
+                {
+                    Program.Error(new CompilerError(null, $"Unknown label {label.Identifier}"));
+                }
+                var labelList = _labels.Keys.ToList();
+                var continueLabel = string.Empty;
+                for (var i = labelList.IndexOf(codeLabel) + 2; i < labelList.Count; i++)
```

Comment:
For the previous suggestion
```suggestion
                for (var i = labelList.IndexOf(codeLabel) + 1; i < labelList.Count; i++)
```